### PR TITLE
Address security concern etld dapps

### PR DESCRIPTION
**Pull request by qamarngr**
Resolves brave/brave-browser#21798

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Same as 
https://github.com/brave/brave-core/pull/12701

### CODE REVIEW

**Review by Pavneet-Sing — COMMENTED** [review]
I would encourage to start using the domain layer (any possible implementation) with MVVM style to enhance the code quality, classes like Utils is doing too much, yet not efficient. Feel free to use your architectural skills to enhance domain or other areas.

**Comment by SergeyZhukovsky** [comment]
@qamarngr I don't think CI is running btw if you work off your fork. As I see lint format isn't applied as well. Could you please work directly from brave-core repository?

File: android/java/org/chromium/chrome/browser/crypto_wallet/permission/BraveEthereumPermissionPromptDialog.java [hunk 1]
```diff
@@ -111,6 +118,13 @@ void show() {
         }
     }
 
+    private void InitBraveWalletService() {
+        if (mBraveWalletService != null) {
+            return;
+        }
+        mBraveWalletService = BraveWalletServiceFactory.getInstance().getBraveWalletService(this);
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/permission/BraveEthereumPermissionPromptDialog.java:125]: we need to close the service in `DisconnectMojoServices`

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/ConnectAccountFragment.java [hunk 2]
```diff
@@ -112,7 +112,9 @@ public void onClick(View v) {
         mRecyclerView = view.findViewById(R.id.accounts_list);
         mFavicon = view.findViewById(R.id.favicon);
 
-        mWebSite.setText(getCurrentHostHttpAddress().getSpec());
+        getBraveWalletService().geteTldPlusOneFromOrigin(Utils.getCurrentMojomOrigin(), origin -> {
+                mWebSite.setText(Utils.geteTLDFromGRUL(origin.eTldPlusOne));
```
[the remark this answers is not in the document]
Comment (qamarngr) [reply android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/ConnectAccountFragment.java:116]: oh sorry, I refactored it, I was previously passing GURL, later on I removed it.

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/DAppsMessageFragment.java [hunk 3]
```diff
@@ -27,7 +29,11 @@ public View onCreateView(
             LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.fragment_dapps_message, container, false);
         TextView signMessageText = view.findViewById(R.id.sign_message_text);
-        signMessageText.setText(mCurrentSignMessageRequest.message);
+        SpannableStringBuilder spBuilder = new SpannableStringBuilder();
+        spBuilder.append(Utils.geteTLDFromGRUL(mCurrentSignMessageRequest.originInfo.eTldPlusOne));
+        spBuilder.append(" ");
+        spBuilder.append(getString(R.string.brave_wallet_provide_encryption_key_description));
```
[the remark this answers is not in the document]
Comment (qamarngr) [reply android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/DAppsMessageFragment.java:35]: Actually `Utils.geteTLDFromGRUL(mCurrentSignMessageRequest.originInfo.eTldPlusOne)` returns Spanned, not a string , can I still template string ? On all other places in code only `geteTLDFromGRUL` is used, here we need to merge/append two `spanned` strings that's why I used this.

**Comment by Pavneet-Sing** [comment]
Duplicate of https://github.com/brave/brave-core/pull/13398

**Pull request closed without merging**